rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/attachment/Sw360AttachmentService.java

```diff
@@ -186,4 +187,10 @@ private synchronized void makeConnector() throws TException {
         }
         return new Resources<>(attachmentResources);
     }
+
+    public boolean isDuplicateAttachment(Set<Attachment> attachments) {
+        boolean duplicateSha1 = attachments.parallelStream().collect(Collectors.groupingBy(Attachment::getSha1)).size() < attachments.size();
```

Comment:
Maybe extract function into 
```
boolean checkForDuplicatesOn(Collection<Attachment> attachments, Function<Attachment,Object> extractor)
```

Reply:
Might be easier to read as:
```suggestion
        boolean duplicateSha1 = attachments.stream().map(Attachment::getSha1).distinct().size() < attachments.size();
```